### PR TITLE
Add Jupyter nbextension library for research branch data visualization

### PR DESCRIPTION
**Pull request by mortonanalytics**
This pull request adds the infrastructure we'll be using for Jupyter Notebook work involving data visualization. The PR adds a library to the `infra` folder and can be installed in a local Python environment (some basic install instructions are in the read me) that works as a Notebook extension (`nbextension` in  python speak). This PR is needed as a foundation element to the research environment where graph objects can be better explored visually.

The library creates communication between the Jupyter/Python environment and the browser. There is a python based set of scripts and JavaScript based scripts. Both code bases may be updated separately as needed.

There is a `notebook_examples` folder that demos the basic usage of the library (though I intend to write a more thorough technical vignette down the road). @mzargham, this notebook example is a good place to start in playing around with the library.

This PR also adds the EDA written on the graph object in the sample-graph folder. This is in the `exploratory/rMorton` folder and is more of an early research project.

### CODE REVIEW

File: exploratory/rMorton/.Rproj.user/C5F3700/build_options [hunk 1]
```diff
@@ -0,0 +1,7 @@
+auto_roxygenize_for_build_and_reload="0"
```
Comment (teamdandelion) [inline exploratory/rMorton/.Rproj.user/C5F3700/build_options:1]: Should this file be tracked in git? Looks like something that should be git ignored.

File: infra/pySourceCredChart/js/lib/example.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+var widgets = require('@jupyter-widgets/base');
```
Comment (teamdandelion) [inline infra/pySourceCredChart/js/lib/example.js]: This looks like a leftover file from a hello-world example? remove?

File: infra/pySourceCredChart/js/lib/pySourceCredChart.js [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+		this.data = opts.data;
+		this.scores = opts.scores;
+		this.mapping = opts.mapping;
+		this.options = opts.options;
+		console.log(this.data);
+		console.log(this.mapping);
+		
+		//preserve context for closures
+		var that = this;
+		/*
```
Comment (teamdandelion) [inline infra/pySourceCredChart/js/lib/pySourceCredChart.js]: let's clean out these large blocks of commented code

File: infra/pySourceCredChart/setup.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+])
+
+from distutils import log
+log.set_verbosity(log.DEBUG)
+log.info('setup.py entered')
+log.info('$PATH=%s' % os.environ['PATH'])
+
+LONG_DESCRIPTION = 'A python library for Source Cred Charts'
+
+def js_prerelease(command, strict=False):
```
Comment (teamdandelion) [inline infra/pySourceCredChart/setup.py:28]: was this code copy/pasted from somewhere? if so can we (ideally) import it (if not) attribute it?

**Review by teamdandelion — COMMENTED** [review]
This PR could use some cleanup, along the lines of removing large commented blocks of code, and ensuring that all the files actually need to get committed (it looks like some .Rproj are local editor state)

Also, could we split this up into 3 PRS:
1. that makes the changes to exploratory/ with the EDA stuff (not sure if it depends on the other 2)
2. that adds the JS package for graph visualization
3. that adds the python interop on top of 2 with examples (and is using 2. as a base branch)

If splitting up the PRs is a challenge for you from a git perspective, you can invoke [`git-rescued`](https://discourse.sourcecred.io/t/git-rescued-a-new-service-for-sourcecred-contributors/100)

**Comment by mortonanalytics** [comment]
@decentralion I can close this pull request and break things up with separate PRs.  EDIT: or maybe I've salvaged this PR as just the python code...

File: infra/pySourceCredChart/js/lib/pySourceCredChart.js (hunk 3 above)
Comment (mortonanalytics) [reply infra/pySourceCredChart/js/lib/pySourceCredChart.js]: Yup

File: infra/pySourceCredChart/js/lib/example.js (hunk 2 above)
Comment (mortonanalytics) [reply infra/pySourceCredChart/js/lib/example.js]: removed

File: exploratory/rMorton/.Rproj.user/C5F3700/build_options (hunk 1 above)
Comment (mortonanalytics) [reply exploratory/rMorton/.Rproj.user/C5F3700/build_options:1]: It depends. If someone wanted to take over the project at where is was, this Rproj file gets them to precise place in the project. I can just gitignore though.

File: infra/pySourceCredChart/setup.py (hunk 4 above)
Comment (mortonanalytics) [reply infra/pySourceCredChart/setup.py:28]: cookiecutter was used for creating the ipywidget. I'll put that in the readme file and this particular file. 

The setup file is for anyone that is developing the python library. It needs to go through this setup.py to make sure all the dependencies are loaded and the py/JS communication is valid.  EDITED: The file is necessary to build (and pip install it, I believe upon closer review) the library, but not use it.

**Review by teamdandelion — APPROVED** [review]
Cool. As with the other PR, I don't have the bandwidth to go super deep here, so I'll approve to keep research unblocked. Let me know if there's anything I should be looking at closely.

**Comment by mortonanalytics** [comment]
Totally fair. Until we start tweaking other aspects, the most important files are just the two pySourceCredChart files (.py and .js). Everything else is the recommended Jupyter Notebook extension framework that we probably don't want to change until we (I?) learn more about it and how changing it may improve or streamline things.

**Pull request merged**